Can not compile anymore with Qt6

Running `sudo make install` on latest libkiwix build outputs:

```
g++ -std=c++17 -Wl,-O1 -Wl,-rpath-link,/usr/lib/x86_64-linux-gnu -o kiwix-desktop choiceitem.o contentmanagerdelegate.o contentmanagerheader.o contentmanagermodel.o contenttypefilter.o descriptionnode.o downloadmanagement.o findinpagebar.o flowlayout.o kiwixchoicebox.o kiwixconfirmbox.o kiwixlineedit.o kiwixlistwidget.o kiwixloader.o rownode.o suggestionlistworker.o thumbnaildownloader.o translation.o main.o mainwindow.o kiwixapp.o kprofile.o blobbuffer.o library.o settingsmanager.o settingsview.o topwidget.o urlschemehandler.o webview.o searchbar.o mainmenu.o webpage.o about.o contentmanager.o contentmanagerview.o tabbar.o contentmanagerside.o readinglistbar.o klistwidgetitem.o opdsrequestmanager.o localkiwixserver.o fullscreenwindow.o fullscreennotification.o zimview.o qtsingleapplication.o qtlocalpeer.o qrc_kiwix.o qrc_translations.o qrc_contentmanager.o qrc_settingsmanager.o qrc_style.o moc_choiceitem.o moc_contentmanagerdelegate.o moc_contentmanagerheader.o moc_contentmanagermodel.o moc_contentmanagerview.o moc_contenttypefilter.o moc_downloadmanagement.o moc_findinpagebar.o moc_flowlayout.o moc_kiwixchoicebox.o moc_kiwixconfirmbox.o moc_kiwixlineedit.o moc_kiwixlistwidget.o moc_kiwixloader.o moc_suggestionlistworker.o moc_thumbnaildownloader.o moc_mainwindow.o moc_kiwixapp.o moc_kprofile.o moc_blobbuffer.o moc_library.o moc_settingsmanager.o moc_settingsview.o moc_topwidget.o moc_urlschemehandler.o moc_webview.o moc_searchbar.o moc_mainmenu.o moc_webpage.o moc_about.o moc_contentmanager.o moc_tabbar.o moc_contentmanagerside.o moc_readinglistbar.o moc_opdsrequestmanager.o moc_localkiwixserver.o moc_fullscreenwindow.o moc_fullscreennotification.o moc_menuproxystyle.o moc_zimview.o moc_qtsingleapplication.o moc_qtlocalpeer.o   /usr/lib/x86_64-linux-gnu/libQt5WebEngineWidgets.so /usr/lib/x86_64-linux-gnu/libQt5WebEngineCore.so /usr/lib/x86_64-linux-gnu/libQt5Quick.so /usr/lib/x86_64-linux-gnu/libQt5PrintSupport.so /usr/lib/x86_64-linux-gnu/libQt5Widgets.so /usr/lib/x86_64-linux-gnu/libQt5Gui.so /usr/lib/x86_64-linux-gnu/libQt5WebChannel.so /usr/lib/x86_64-linux-gnu/libQt5Qml.so /usr/lib/x86_64-linux-gnu/libQt5Network.so /usr/lib/x86_64-linux-gnu/libQt5Positioning.so /usr/lib/x86_64-linux-gnu/libQt5Core.so /usr/lib/x86_64-linux-gnu/libGL.so -lpthread
/usr/bin/ld: contentmanagermodel.o: in function `ContentManagerModel::createNode(QMap<QString, QVariant>) const':
contentmanagermodel.cpp:(.text+0xbfa): undefined reference to `kiwix::beautifyFileSize[abi:cxx11](unsigned long)'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::resumeDownload(QString const&)':
downloadmanagement.cpp:(.text+0xa13): undefined reference to `kiwix::Downloader::getDownload(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: downloadmanagement.cpp:(.text+0xad9): undefined reference to `kiwix::Download::resumeDownload()'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::startDownload[abi:cxx11](kiwix::Book const&, QString const&)':
downloadmanagement.cpp:(.text+0xc69): undefined reference to `kiwix::Downloader::startDownload(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >)'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::getDownloadInfo(QString) const':
downloadmanagement.cpp:(.text+0x1cff): undefined reference to `kiwix::Downloader::getDownload(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: downloadmanagement.cpp:(.text+0x1d0e): undefined reference to `kiwix::Download::updateStatus(bool)'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::pauseDownload(QString const&)':
downloadmanagement.cpp:(.text+0x28aa): undefined reference to `kiwix::Downloader::getDownload(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: downloadmanagement.cpp:(.text+0x28e1): undefined reference to `kiwix::Download::pauseDownload()'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::cancelDownload(QString const&)':
downloadmanagement.cpp:(.text+0x2a0a): undefined reference to `kiwix::Downloader::getDownload(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: downloadmanagement.cpp:(.text+0x2a14): undefined reference to `kiwix::Download::cancelDownload()'
/usr/bin/ld: downloadmanagement.o: in function `DownloadManager::DownloadManager(Library const*)':
downloadmanagement.cpp:(.text+0x3337): undefined reference to `kiwix::Downloader::Downloader(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/usr/bin/ld: suggestionlistworker.o: in function `SuggestionListWorker::run()':
suggestionlistworker.cpp:(.text+0x48d): undefined reference to `zim::SuggestionSearcher::SuggestionSearcher(zim::Archive const&)'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x4ad): undefined reference to `zim::SuggestionSearcher::suggest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x4c4): undefined reference to `zim::SuggestionSearch::getResults(int, int) const'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x4e1): undefined reference to `zim::SuggestionResultSet::begin() const'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x4ee): undefined reference to `zim::SuggestionResultSet::end() const'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x65d): undefined reference to `zim::SuggestionIterator::operator++()'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x66a): undefined reference to `zim::SuggestionIterator::operator!=(zim::SuggestionIterator const&) const'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x67a): undefined reference to `zim::SuggestionIterator::operator*()'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x999): undefined reference to `zim::SuggestionIterator::~SuggestionIterator()'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x9a1): undefined reference to `zim::SuggestionIterator::~SuggestionIterator()'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0x9fd): undefined reference to `zim::Archive::hasFulltextIndex() const'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0xe56): undefined reference to `zim::SuggestionSearch::~SuggestionSearch()'
/usr/bin/ld: suggestionlistworker.cpp:(.text+0xe60): undefined reference to `zim::SuggestionSearcher::~SuggestionSearcher()'
/usr/bin/ld: suggestionlistworker.o: in function `SuggestionListWorker::run() [clone .cold]':
suggestionlistworker.cpp:(.text.unlikely+0x72): undefined reference to `zim::SuggestionSearch::~SuggestionSearch()'
/usr/bin/ld: suggestionlistworker.cpp:(.text.unlikely+0x7c): undefined reference to `zim::SuggestionSearcher::~SuggestionSearcher()'
/usr/bin/ld: suggestionlistworker.cpp:(.text.unlikely+0x100): undefined reference to `zim::SuggestionIterator::~SuggestionIterator()'
/usr/bin/ld: suggestionlistworker.cpp:(.text.unlikely+0x108): undefined reference to `zim::SuggestionIterator::~SuggestionIterator()'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::isCurrentArticleBookmarked()':
kiwixapp.cpp:(.text+0xc19): undefined reference to `kiwix::Library::getBookmarks(bool) const'
/usr/bin/ld: kiwixapp.cpp:(.text+0xcbb): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: kiwixapp.cpp:(.text+0xdcb): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::printVersions(std::ostream&)':
kiwixapp.cpp:(.text+0x12a0): undefined reference to `kiwix::printVersions(std::ostream&)'
/usr/bin/ld: kiwixapp.cpp:(.text+0x12e3): undefined reference to `zim::printVersions(std::ostream&)'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::findLibraryDirectory()':
kiwixapp.cpp:(.text+0x2726): undefined reference to `kiwix::getExecutablePath[abi:cxx11](bool)'
/usr/bin/ld: kiwixapp.cpp:(.text+0x273d): undefined reference to `kiwix::removeLastPathElement(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::KiwixApp(int&, char**)':
kiwixapp.cpp:(.text+0x30b2): undefined reference to `kiwix::UpdatableNameMapper::UpdatableNameMapper(std::shared_ptr<kiwix::Library>, bool)'
/usr/bin/ld: kiwixapp.cpp:(.text+0x3159): undefined reference to `kiwix::Server::Server(std::shared_ptr<kiwix::Library>, std::shared_ptr<kiwix::NameMapper>)'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::~KiwixApp()':
kiwixapp.cpp:(.text+0x387e): undefined reference to `kiwix::Server::stop()'
/usr/bin/ld: kiwixapp.cpp:(.text+0x38ce): undefined reference to `kiwix::Server::~Server()'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::openRandomUrl(bool)':
kiwixapp.cpp:(.text+0x5635): undefined reference to `zim::Archive::getRandomEntry() const'
/usr/bin/ld: kiwixapp.cpp:(.text+0x572d): undefined reference to `zim::Entry::getPath[abi:cxx11]() const'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::updateNameMapper()':
kiwixapp.cpp:(.text+0x8c): undefined reference to `kiwix::UpdatableNameMapper::update()'
/usr/bin/ld: kiwixapp.o: in function `KiwixApp::KiwixApp(int&, char**) [clone .cold]':
kiwixapp.cpp:(.text.unlikely+0x525): undefined reference to `kiwix::Server::~Server()'
/usr/bin/ld: library.o: in function `Library::addBookmark(kiwix::Bookmark&)':
library.cpp:(.text+0x8d): undefined reference to `kiwix::Library::addBookmark(kiwix::Bookmark const&)'
/usr/bin/ld: library.o: in function `Library::readBookMarksFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
library.cpp:(.text+0xfb): undefined reference to `kiwix::Manager::Manager(std::shared_ptr<kiwix::Library>)'
/usr/bin/ld: library.cpp:(.text+0x137): undefined reference to `kiwix::Manager::readBookmarkFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x14c): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.o: in function `Library::getBookIds() const':
library.cpp:(.text+0x26c): undefined reference to `kiwix::Library::getBooksIds[abi:cxx11]() const'
/usr/bin/ld: library.o: in function `Library::listBookIds(kiwix::Filter const&, kiwix::supportedListSortBy, bool) const':
library.cpp:(.text+0x3d5): undefined reference to `kiwix::Library::filter[abi:cxx11](kiwix::Filter const&) const'
/usr/bin/ld: library.cpp:(.text+0x3e7): undefined reference to `kiwix::Library::sort(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, kiwix::supportedListSortBy, bool) const'
/usr/bin/ld: library.o: in function `Library::removeBookFromLibraryById(QString const&)':
library.cpp:(.text+0x588): undefined reference to `kiwix::Library::removeBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.o: in function `Library::getBookById(QString) const':
library.cpp:(.text+0x668): undefined reference to `kiwix::Library::getBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: library.o: in function `Library::removeBookmark(QString const&, QString const&)':
library.cpp:(.text+0xcc8): undefined reference to `kiwix::Library::removeBookmark(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.o: in function `Library::getArchive(QString const&)':
library.cpp:(.text+0xe06): undefined reference to `kiwix::Library::getArchiveById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.o: in function `Library::save()':
library.cpp:(.text+0xf4d): undefined reference to `kiwix::appendToDirectory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0xf5a): undefined reference to `kiwix::Library::writeToFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: library.cpp:(.text+0x1015): undefined reference to `kiwix::appendToDirectory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x1022): undefined reference to `kiwix::Library::writeBookmarksToFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: library.o: in function `Library::Library(QString const&)':
library.cpp:(.text+0x1275): undefined reference to `kiwix::Library::Library()'
/usr/bin/ld: library.cpp:(.text+0x1329): undefined reference to `kiwix::LibraryManipulator::LibraryManipulator(std::shared_ptr<kiwix::Library>)'
/usr/bin/ld: library.cpp:(.text+0x1378): undefined reference to `vtable for kiwix::LibraryManipulator'
/usr/bin/ld: library.cpp:(.text+0x13c0): undefined reference to `kiwix::Manager::Manager(kiwix::LibraryManipulator)'
/usr/bin/ld: library.cpp:(.text+0x13d0): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.cpp:(.text+0x13e1): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.cpp:(.text+0x1480): undefined reference to `kiwix::appendToDirectory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x1492): undefined reference to `kiwix::Manager::readFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)'
/usr/bin/ld: library.cpp:(.text+0x1555): undefined reference to `kiwix::appendToDirectory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x1560): undefined reference to `kiwix::Manager::readBookmarkFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x15ae): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.o: in function `Library::openBookFromPath(QString const&)':
library.cpp:(.text+0x187d): undefined reference to `kiwix::Library::getBookByPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: library.o: in function `Library::updateBookBeingDownloaded(QString const&, QString const&)':
library.cpp:(.text+0x1d64): undefined reference to `kiwix::Book::setPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x1d71): undefined reference to `kiwix::Library::addBook(kiwix::Book const&)'
/usr/bin/ld: library.cpp:(.text+0x1d81): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: library.o: in function `Library::isBeingDownloadedByUs(QString) const':
library.cpp:(.text+0x1f49): undefined reference to `kiwix::Library::getBookByPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: library.o: in function `Library::addBookBeingDownloaded(kiwix::Book const&, QString)':
library.cpp:(.text+0x21bc): undefined reference to `kiwix::Book::setPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: library.cpp:(.text+0x21df): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: library.o: in function `Library::getSearcher(QString const&)':
library.cpp:(.text+0x2541): undefined reference to `kiwix::Library::getSearcherByIds(std::set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
/usr/bin/ld: library.o: in function `Library::addBookToLibrary(kiwix::Book&)':
library.cpp:(.text+0x79): undefined reference to `kiwix::Library::addBook(kiwix::Book const&)'
/usr/bin/ld: library.o: in function `std::_Sp_counted_ptr<kiwix::Library*, (__gnu_cxx::_Lock_policy)2>::_M_dispose()':
library.cpp:(.text._ZNSt15_Sp_counted_ptrIPN5kiwix7LibraryELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt15_Sp_counted_ptrIPN5kiwix7LibraryELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x12): undefined reference to `kiwix::Library::~Library()'
/usr/bin/ld: library.o: in function `LibraryManipulator::~LibraryManipulator()':
library.cpp:(.text._ZN18LibraryManipulatorD2Ev[_ZN18LibraryManipulatorD5Ev]+0x13): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.o: in function `LibraryManipulator::~LibraryManipulator()':
library.cpp:(.text._ZN18LibraryManipulatorD0Ev[_ZN18LibraryManipulatorD5Ev]+0x17): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.o: in function `kiwix::Manager::~Manager()':
library.cpp:(.text._ZN5kiwix7ManagerD2Ev[_ZN5kiwix7ManagerD5Ev]+0x11): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.o: in function `Library::Library(QString const&) [clone .cold]':
library.cpp:(.text.unlikely+0x2ec): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.cpp:(.text.unlikely+0x300): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: library.cpp:(.text.unlikely+0x362): undefined reference to `kiwix::Library::~Library()'
/usr/bin/ld: library.o: in function `Library::openBookFromPath(QString const&) [clone .cold]':
library.cpp:(.text.unlikely+0x437): undefined reference to `kiwix::Manager::Manager(std::shared_ptr<kiwix::Library>)'
/usr/bin/ld: library.cpp:(.text.unlikely+0x4c9): undefined reference to `kiwix::Manager::addBookFromPathAndGetId(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/bin/ld: library.o: in function `Library::updateBookBeingDownloaded(QString const&, QString const&) [clone .cold]':
library.cpp:(.text.unlikely+0x747): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: library.o: in function `Library::addBookBeingDownloaded(kiwix::Book const&, QString) [clone .cold]':
library.cpp:(.text.unlikely+0x7dd): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: library.o:(.data.rel.ro._ZTI18LibraryManipulator[_ZTI18LibraryManipulator]+0x10): undefined reference to `typeinfo for kiwix::LibraryManipulator'
/usr/bin/ld: library.o:(.data.rel.ro._ZTV18LibraryManipulator[_ZTV18LibraryManipulator]+0x20): undefined reference to `kiwix::LibraryManipulator::bookWasAddedToLibrary(kiwix::Book const&)'
/usr/bin/ld: library.o:(.data.rel.ro._ZTV18LibraryManipulator[_ZTV18LibraryManipulator]+0x28): undefined reference to `kiwix::LibraryManipulator::bookmarkWasAddedToLibrary(kiwix::Bookmark const&)'
/usr/bin/ld: library.o:(.data.rel.ro._ZTV18LibraryManipulator[_ZTV18LibraryManipulator]+0x30): undefined reference to `kiwix::LibraryManipulator::booksWereRemovedFromLibrary()'
/usr/bin/ld: settingsmanager.o: in function `getDataDirectory()':
settingsmanager.cpp:(.text+0x24b): undefined reference to `kiwix::getCurrentDirectory[abi:cxx11]()'
/usr/bin/ld: urlschemehandler.o: in function `getArchiveEntryFromUrl(zim::Archive const&, QUrl const&)':
urlschemehandler.cpp:(.text+0x1d6): undefined reference to `zim::Archive::getEntryByPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: urlschemehandler.o: in function `UrlSchemeHandler::handleSearchRequest(QWebEngineUrlRequestJob*)':
urlschemehandler.cpp:(.text+0x920): undefined reference to `zim::Query::Query(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x935): undefined reference to `zim::Searcher::search(zim::Query const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x974): undefined reference to `zim::Search::Search(zim::Search&&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x97c): undefined reference to `zim::Search::~Search()'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x9d4): undefined reference to `zim::Search::getEstimatedMatches() const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x9f2): undefined reference to `zim::Search::getResults(int, int) const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xb4c): undefined reference to `kiwix::SearchRenderer::SearchRenderer(zim::SearchResultSet, unsigned int, unsigned int)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xbbb): undefined reference to `kiwix::SearchRenderer::setSearchPattern(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xc99): undefined reference to `kiwix::SearchRenderer::setSearchBookQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xcef): undefined reference to `kiwix::SearchRenderer::setProtocolPrefix(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xe5f): undefined reference to `kiwix::SearchRenderer::setSearchProtocolPrefix(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0xecd): undefined reference to `kiwix::SearchRenderer::getHtml[abi:cxx11](kiwix::NameMapper const&, kiwix::Library const*)'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x1047): undefined reference to `kiwix::SearchRenderer::~SearchRenderer()'
/usr/bin/ld: urlschemehandler.o: in function `UrlSchemeHandler::handleMetaRequest(QWebEngineUrlRequestJob*)':
urlschemehandler.cpp:(.text+0x1b62): undefined reference to `kiwix::Book::getIllustration(unsigned int) const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x1b6e): undefined reference to `kiwix::Book::Illustration::getData[abi:cxx11]() const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x1dd8): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: urlschemehandler.o: in function `UrlSchemeHandler::handleContentRequest(QWebEngineUrlRequestJob*)':
urlschemehandler.cpp:(.text+0x533b): undefined reference to `zim::Entry::getItem(bool) const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x5343): undefined reference to `zim::Entry::isRedirect() const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x535f): undefined reference to `zim::Item::getData(unsigned long) const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x53e8): undefined reference to `zim::Item::getMimetype[abi:cxx11]() const'
/usr/bin/ld: urlschemehandler.cpp:(.text+0x5848): undefined reference to `zim::Entry::getPath[abi:cxx11]() const'
/usr/bin/ld: urlschemehandler.o: in function `std::_Sp_counted_ptr_inplace<zim::Search, std::allocator<zim::Search>, (__gnu_cxx::_Lock_policy)2>::_M_dispose()':
urlschemehandler.cpp:(.text._ZNSt23_Sp_counted_ptr_inplaceIN3zim6SearchESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt23_Sp_counted_ptr_inplaceIN3zim6SearchESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x9): undefined reference to `zim::Search::~Search()'
/usr/bin/ld: urlschemehandler.o: in function `getArchiveEntryFromUrl(zim::Archive const&, QUrl const&) [clone .cold]':
urlschemehandler.cpp:(.text.unlikely+0x32): undefined reference to `zim::Archive::getMainEntry() const'
/usr/bin/ld: urlschemehandler.o: in function `UrlSchemeHandler::handleSearchRequest(QWebEngineUrlRequestJob*) [clone .cold]':
urlschemehandler.cpp:(.text.unlikely+0x19c): undefined reference to `zim::Search::~Search()'
/usr/bin/ld: urlschemehandler.cpp:(.text.unlikely+0x274): undefined reference to `kiwix::SearchRenderer::~SearchRenderer()'
/usr/bin/ld: urlschemehandler.o: in function `UrlSchemeHandler::handleMetaRequest(QWebEngineUrlRequestJob*) [clone .cold]':
urlschemehandler.cpp:(.text.unlikely+0x4e7): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: webview.o: in function `WebView::saveViewContent()':
webview.cpp:(.text+0x1806): undefined reference to `zim::Entry::isRedirect() const'
/usr/bin/ld: webview.cpp:(.text+0x181e): undefined reference to `zim::Entry::getItem(bool) const'
/usr/bin/ld: webview.cpp:(.text+0x1836): undefined reference to `zim::Item::getMimetype[abi:cxx11]() const'
/usr/bin/ld: webview.cpp:(.text+0x18c6): undefined reference to `zim::Entry::getTitle[abi:cxx11]() const'
/usr/bin/ld: webview.cpp:(.text+0x18d3): undefined reference to `kiwix::getSlugifiedFileName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: webview.o: in function `WebView::onUrlChanged(QUrl const&)':
webview.cpp:(.text+0x2700): undefined reference to `zim::Archive::getIllustrationItem(unsigned int) const'
/usr/bin/ld: webview.cpp:(.text+0x2712): undefined reference to `zim::Item::getData(unsigned long) const'
/usr/bin/ld: webview.cpp:(.text+0x2845): undefined reference to `zim::Item::getMimetype[abi:cxx11]() const'
/usr/bin/ld: searchbar.o: in function `BookmarkButton::on_buttonClicked()':
searchbar.cpp:(.text+0x5f1): undefined reference to `kiwix::Bookmark::Bookmark()'
/usr/bin/ld: searchbar.cpp:(.text+0x87e): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: searchbar.o: in function `BookmarkButton::on_buttonClicked() [clone .cold]':
searchbar.cpp:(.text.unlikely+0x64): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::eraseBookFilesFromComputer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)':
contentmanager.cpp:(.text+0x2004): undefined reference to `kiwix::removeLastPathElement(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x200f): undefined reference to `kiwix::getLastPathElement(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::getRemoteOrLocalBook(QString const&)':
contentmanager.cpp:(.text+0x38b0): undefined reference to `kiwix::Library::getBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: contentmanager.o: in function `ContentManager::getBookIds()':
contentmanager.cpp:(.text+0x396e): undefined reference to `kiwix::Filter::Filter()'
/usr/bin/ld: contentmanager.cpp:(.text+0x3ac3): undefined reference to `kiwix::Filter::acceptTags(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3ad3): undefined reference to `kiwix::Filter::rejectTags(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3afb): undefined reference to `kiwix::Filter::query(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3b5a): undefined reference to `kiwix::Filter::lang(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3bb2): undefined reference to `kiwix::Filter::local(bool)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3bbf): undefined reference to `kiwix::Filter::valid(bool)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3c31): undefined reference to `kiwix::Filter::remote(bool)'
/usr/bin/ld: contentmanager.cpp:(.text+0x3c60): undefined reference to `kiwix::Library::filter[abi:cxx11](kiwix::Filter const&) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x3c7e): undefined reference to `kiwix::Library::sort(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, kiwix::supportedListSortBy, bool) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x3d37): undefined reference to `kiwix::Filter::category(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::openBookPreview(QString const&)':
contentmanager.cpp:(.text+0x3e32): undefined reference to `kiwix::Library::getBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadDisappeared(QString)':
contentmanager.cpp:(.text+0x409a): undefined reference to `kiwix::Book::Book()'
/usr/bin/ld: contentmanager.cpp:(.text+0x4296): undefined reference to `kiwix::Library::addBook(kiwix::Book const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x42bd): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::updateLanguages(QString const&)':
contentmanager.cpp:(.text+0x4503): undefined reference to `kiwix::readLanguagesFromFeed(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::setCategories()':
contentmanager.cpp:(.text+0x4855): undefined reference to `kiwix::Library::getBooksCategories[abi:cxx11]() const'
/usr/bin/ld: contentmanager.o: in function `ContentManager::setLanguages()':
contentmanager.cpp:(.text+0x4a98): undefined reference to `kiwix::Library::getBooksLanguages[abi:cxx11]() const'
/usr/bin/ld: contentmanager.cpp:(.text+0x4b28): undefined reference to `kiwix::getLanguageSelfName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::updateCategories(QString const&)':
contentmanager.cpp:(.text+0x4d98): undefined reference to `kiwix::readCategoriesFromFeed(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadCompleted(QString, QString)':
contentmanager.cpp:(.text+0x4ffc): undefined reference to `kiwix::Book::setPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x50b7): undefined reference to `kiwix::Library::addBook(kiwix::Book const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0x50f2): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadBook(QString const&)':
contentmanager.cpp:(.text+0x5672): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::startDownload(QString)':
contentmanager.cpp:(.text+0x58cd): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::getBookInfos(QString, QStringList const&)':
contentmanager.cpp:(.text+0x5a47): undefined reference to `kiwix::Library::getBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x639f): undefined reference to `kiwix::Book::getIllustration(unsigned int) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x6461): undefined reference to `kiwix::Book::getIllustration(unsigned int) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x646b): undefined reference to `kiwix::Book::Illustration::getData[abi:cxx11]() const'
/usr/bin/ld: contentmanager.o: in function `ContentManager::setCurrentLanguage(QList<QPair<QString, QString> >)':
contentmanager.cpp:(.text+0x790a): undefined reference to `kiwix::converta2toa3(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.o: in function `ContentManager::handleDisappearedZimFiles(QString const&, QSet<QString> const&)':
contentmanager.cpp:(.text+0x7fc4): undefined reference to `kiwix::Library::getBookByPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: contentmanager.cpp:(.text+0x8039): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::ContentManager(Library*)':
contentmanager.cpp:(.text+0x836c): undefined reference to `kiwix::Library::Library()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::onCustomContextMenu(QPoint const&)':
contentmanager.cpp:(.text+0x9f2f): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `QtConcurrent::StoredFunctorCall0<void, ContentManager::updateRemoteLibrary(QString const&)::{lambda()#1}>::runFunctor()':
contentmanager.cpp:(.text+0xa27f): undefined reference to `kiwix::Library::Library()'
/usr/bin/ld: contentmanager.cpp:(.text+0xa314): undefined reference to `kiwix::Manager::Manager(std::shared_ptr<kiwix::Library>)'
/usr/bin/ld: contentmanager.cpp:(.text+0xa35e): undefined reference to `kiwix::Manager::readOpds(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: contentmanager.cpp:(.text+0xa3af): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::handleZimFileInMonitoredDir(QString, QString)':
contentmanager.cpp:(.text+0xa6c3): undefined reference to `kiwix::Manager::Manager(std::shared_ptr<kiwix::Library>)'
/usr/bin/ld: contentmanager.cpp:(.text+0xa764): undefined reference to `kiwix::Manager::addBookFromPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/bin/ld: contentmanager.cpp:(.text+0xa7dc): undefined reference to `kiwix::LibraryManipulator::~LibraryManipulator()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::getBookState(QString) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0x6bb): undefined reference to `kiwix::Library::getBookById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadDisappeared(QString) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0x911): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.cpp:(.text.unlikely+0x93c): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadCompleted(QString, QString) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0xbd5): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::downloadBook(QString const&) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0xcf0): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.cpp:(.text.unlikely+0xd3a): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o:contentmanager.cpp:(.text.unlikely+0xd78): more undefined references to `kiwix::Book::~Book()' follow
/usr/bin/ld: contentmanager.o: in function `ContentManager::ContentManager(Library*) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0x141e): undefined reference to `kiwix::Library::~Library()'
/usr/bin/ld: contentmanager.o: in function `ContentManager::onCustomContextMenu(QPoint const&) [clone .cold]':
contentmanager.cpp:(.text.unlikely+0x14d8): undefined reference to `kiwix::Book::~Book()'
/usr/bin/ld: contentmanager.o: in function `QtConcurrent::StoredFunctorCall0<void, ContentManager::updateRemoteLibrary(QString const&)::{lambda()#1}>::runFunctor() [clone .cold]':
contentmanager.cpp:(.text.unlikely+0x16b9): undefined reference to `kiwix::Library::~Library()'
/usr/bin/ld: contentmanagerside.o: in function `(anonymous namespace)::getCategorySelectionList(QStringList)':
contentmanagerside.cpp:(.text+0x925): undefined reference to `kiwix::translateBookCategory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: readinglistbar.o: in function `ReadingListBar::onExport()':
readinglistbar.cpp:(.text+0x2fc): undefined reference to `kiwix::Library::writeBookmarksToFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/bin/ld: readinglistbar.o: in function `ReadingListBar::openUrl(QListWidgetItem*, bool)':
readinglistbar.cpp:(.text+0xd06): undefined reference to `kiwix::Library::getBookmarks(bool) const'
/usr/bin/ld: readinglistbar.cpp:(.text+0xeb3): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: readinglistbar.cpp:(.text+0x1081): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: readinglistbar.o: in function `ReadingListBar::setupList()':
readinglistbar.cpp:(.text+0x13f0): undefined reference to `kiwix::Library::getBookmarks(bool) const'
/usr/bin/ld: readinglistbar.cpp:(.text+0x15e9): undefined reference to `zim::Archive::getIllustrationItem(unsigned int) const'
/usr/bin/ld: readinglistbar.cpp:(.text+0x15f6): undefined reference to `zim::Item::getData(unsigned long) const'
/usr/bin/ld: readinglistbar.cpp:(.text+0x166c): undefined reference to `zim::Item::getMimetype[abi:cxx11]() const'
/usr/bin/ld: readinglistbar.cpp:(.text+0x17bb): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: readinglistbar.o: in function `std::vector<kiwix::Bookmark, std::allocator<kiwix::Bookmark> >::~vector()':
readinglistbar.cpp:(.text._ZNSt6vectorIN5kiwix8BookmarkESaIS1_EED2Ev[_ZNSt6vectorIN5kiwix8BookmarkESaIS1_EED5Ev]+0x2b): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: readinglistbar.o: in function `ReadingListBar::openUrl(QListWidgetItem*, bool) [clone .cold]':
readinglistbar.cpp:(.text.unlikely+0x12e): undefined reference to `kiwix::Bookmark::~Bookmark()'
/usr/bin/ld: localkiwixserver.o: in function `LocalKiwixServer::LocalKiwixServer(QWidget*)':
localkiwixserver.cpp:(.text+0xaae): undefined reference to `kiwix::getNetworkInterfacesIPv4Or6[abi:cxx11]()'
/usr/bin/ld: localkiwixserver.o: in function `LocalKiwixServer::runOrStopServer()':
localkiwixserver.cpp:(.text+0x1c1c): undefined reference to `kiwix::Server::start()'
/usr/bin/ld: localkiwixserver.cpp:(.text+0x1d4f): undefined reference to `kiwix::Server::stop()'
/usr/bin/ld: localkiwixserver.cpp:(.text+0x1fb0): undefined reference to `kiwix::getBestPublicIp[abi:cxx11](bool)'
collect2: error: ld returned 1 exit status
make: *** [Makefile:432: kiwix-desktop] Error 1
```

`ldconfig -p | grep libkiwix` gives me:
```
libkiwix.so.14 (libc6,x86-64) => /usr/local/lib/x86_64-linux-gnu/libkiwix.so.14
maklibkiwix.so.13 (libc6,x86-64) => /usr/local/lib/x86_64-linux-gnu/libkiwix.so.13
libkiwix.so.3 (libc6,x86-64) => /lib/x86_64-linux-gnu/libkiwix.so.3
libkiwix.so (libc6,x86-64) => /usr/local/lib/x86_64-linux-gnu/libkiwix.so
libkiwix.so (libc6,x86-64) => /lib/x86_64-linux-gnu/libkiwix.so
```

Idk what exactly changed, I did `make clean` and `qmake .` a bunch of times and `sudo make install` now gives:
```
src/contenttypefilter.cpp: In constructor ‘ContentTypeFilter::ContentTypeFilter(QString, QWidget*)’:
src/contenttypefilter.cpp:15:31: error: ‘void QCheckBox::stateChanged(int)’ is deprecated: Use checkStateChanged() instead [-Werror=deprecated-declarations]
   15 |     connect(this, &QCheckBox::stateChanged, this, &ContentTypeFilter::onStateChanged);
      |                               ^~~~~~~~~~~~
In file included from /home/ubunter/Qt/6.8.0/gcc_64/include/QtWidgets/QCheckBox:1,
                 from src/contenttypefilter.h:4,
                 from src/contenttypefilter.cpp:1:
/home/ubunter/Qt/6.8.0/gcc_64/include/QtWidgets/qcheckbox.h:41:10: note: declared here
   41 |     void stateChanged(int);
      |          ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [Makefile:14775: contenttypefilter.o] Error 1
```

--- a/src/contenttypefilter.cpp
+++ b/src/contenttypefilter.cpp
@@ -12,7 +12,11 @@ ContentTypeFilter::ContentTypeFilter(QString name, QWidget *parent)
     m_states[Qt::Checked] = gt("no");
     setText(gt(m_name) + " : " + m_states[checkState()]);
     setStyleSheet("* { color: #666666; }");
-    connect(this, &QCheckBox::stateChanged, this, &ContentTypeFilter::onStateChanged);
+    #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+        connect(this, &QCheckBox::checkStateChanged, this, &ContentTypeFilter::onStateChanged);
+    #else
+        connect(this, &QCheckBox::stateChanged, this, &ContentTypeFilter::onStateChanged);
+    #endif
 }
 
 void ContentTypeFilter::onStateChanged(int state)

--- a/src/flowlayout.cpp
+++ b/src/flowlayout.cpp
@@ -49,6 +49,7 @@
 ****************************************************************************/
 #include "flowlayout.h"
 #include <QtWidgets>
+#include <algorithm>
 //! [1]
 FlowLayout::FlowLayout(QWidget *parent, int margin, int hSpacing, int vSpacing)
     : QLayout(parent), m_hSpace(hSpacing), m_vSpace(vSpacing)
@@ -153,7 +154,7 @@ QSize FlowLayout::sizeHint() const
 QSize FlowLayout::minimumSize() const
 {
     QSize size;
-    for (const QLayoutItem *item : qAsConst(itemList))
+    for (const QLayoutItem *item : std::as_const(itemList))
         size = size.expandedTo(item->minimumSize());
 
     const QMargins margins = contentsMargins();
@@ -174,7 +175,7 @@ int FlowLayout::doLayout(const QRect &rect, bool testOnly) const
 //! [9]
 
 //! [10]
-    for (QLayoutItem *item : qAsConst(itemList)) {
+    for (QLayoutItem *item : std::as_const(itemList)) {
         const QWidget *wid = item->widget();
         int spaceX = horizontalSpacing();
         if (spaceX == -1)

--- a/src/readinglistbar.cpp
+++ b/src/readinglistbar.cpp
@@ -70,16 +70,14 @@ void ReadingListBar::setupList()
             QPixmap pixmap;
             pixmap.loadFromData(reinterpret_cast<const uchar*>(content.data()), content.size());
             auto icon = QIcon(pixmap);
-            auto item = new QListWidgetItem(
+            new QListWidgetItem(
                 icon,
                 QString::fromStdString(bookmark.getTitle()),
                 listWidget);
-            item->setTextAlignment(Qt::TextWordWrap);
         } catch (zim::EntryNotFound& e) {
-            auto item = new QListWidgetItem(
+            new QListWidgetItem(
                 QString::fromStdString(bookmark.getTitle()),
                 listWidget);
-            item->setTextAlignment(Qt::TextWordWrap);
         }
     }
 }

